pkg/watches/watches.go

```diff
@@ -105,3 +108,23 @@ func verifyGVK(gvk schema.GroupVersionKind) error {
 	}
 	return nil
 }
+
+// the template is not here. Should we not provide the template as well for SDK consume it?
+// Should not be part of the lib provide its specific templates as well?
+// This is a point that in my mind has 2 options;
+
+// A) scaffold also be in the libs and libs provide a plugin
+// which means e.g SDK lib = go sdk plugin
+// then, e2e tests for each in the lib repo
+// sdk cli with just commands and calling plugins
+// e.g helm lib is the helm plugin with its testdata and a project scaffolded
+// e.g helm lib/plugin will use sdk lib to do the base
+// we will able to develop and test the changes standalone
+
+// B) scaffolds be part of sdk tool
+// then libs has just the unit test
+// we will need to bump the new lib version to do the full test and check the changes
+// the e2e test are in sdk cli project
+
+// IMO shows that the option A will make easier work with and solve the CI problems
+// and keep better the cohesion.
```

Comment:
just my thought when I was doing this check that I'd like to share with you @joelanford

Reply:
Yep! Just haven't gotten to the plugin implementation yet. The Helm scaffolding plugin will be added to this repository. I started going down that path in a separate branch, but there is a BUNCH of stuff we would need to duplicate or at least pull out of the SDK repo.

As things currently stand, we would need to import from SDK to get some of the common scaffolds, but SDK would need to import from here to get the plugin. That would mean each repo depends on the other, which seems like it would blow up in our faces.

Reply:
My idea here was that all that will be required for the helm plugin from SDK, would come from SDK lib to avoid it.

Reply:
Maybe some of it will come from the SDK lib, but if Kubebuilder is unwilling to open up the scaffolds for the common stuff (e.g. the `./config` directory), maybe we should think twice before doing it. Definitely something to discuss when we're doing the repo split.

Reply:
Just to kept in mind. However, out of scope now. So closing.